### PR TITLE
fix: redirection to learners tab in inContext view

### PR DESCRIPTION
**Pull request by sohailfatima**
[INF-1224](https://2u-internal.atlassian.net/browse/INF-1224)

### Description

Fixed the issue of redirection to the learners tab within the inContext view when username is clicked from post details.
The username in the inContext view is now simple text.

**Before:**

https://github.com/openedx/frontend-app-discussions/assets/68312464/43648b61-a146-4063-967c-d7841a0356fc

**After:**

https://github.com/openedx/frontend-app-discussions/assets/68312464/9182b0e5-0d42-48a1-a150-08aa9f12cc57

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @sohailfatima! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.


:warning: We can't start reviewing your pull request until you've submitted a     [signed contributor agreement](https://openedx.org/cla)     or indicated your institutional affiliation. Please see the [CONTRIBUTING](https://github.com/openedx/.github/blob/master/CONTRIBUTING.md)     file for more information.     If you've signed an agreement in the past, you may need to re-sign.  See     [The New Home of the Open edX Codebase](https://open.edx.org/blog/the-new-home-of-the-open-edx-codebase/)     for details.

    
Once you've signed the CLA, please allow 1 business day for it to be processed.  After this time, you can re-run the     CLA check by adding a comment here that you have signed it.  If the problem persists, you can tag the `@openedx/cla-problems` team in a     comment on your PR for further assistance.

**Comment by mphilbrick211** [comment]
Hi @sohailfatima! Thanks for this contribution! Please let me know if you have any questions regarding submitting a [CLA form](https://openedx.atlassian.net/wiki/spaces/COMM/pages/941457737/How+to+start+contributing+to+the+Open+edX+code+base).

**Comment by ayesha-waris** [comment]
Please consider adding before and after screen recordings in description.

File: src/discussions/common/AuthorLabel.jsx [hunk 1]
```diff
@@ -109,13 +110,21 @@ const AuthorLabel = ({
...
+  const { search } = useLocation();
+
   return showUserNameAsLink
     ? (
       <div className={className}>
         <Link
           data-testid="learner-posts-link"
           id="learner-posts-link"
-          to={generatePath(Routes.LEARNERS.POSTS, { learnerUsername: author, courseId })}
+          to={`${pathname}${enableInContextSidebar ? search : ''}`}
```
Comment (ayesha-waris) [inline src/discussions/common/AuthorLabel.jsx:127]: Post/Comment/Response author’s name should be just simple text in inContext view
[https://2u-internal.atlassian.net/browse/INF-1224](INF-1224)

**Review by ayesha-waris — CHANGES_REQUESTED** [review]
Please implement it according to ticket description

**Review by awais-ansari — CHANGES_REQUESTED** [review]
@sohailfatima The author name should be just simple text in the inContext sidebar view. Currently, implementation is not according to ticket requirements.

**Comment by sohailfatima** [comment]
@awais-ansari @ayesha-waris fixed the issue.

**Review by awais-ansari — CHANGES_REQUESTED** [review]
@sohailfatima, You just need to add logic in this [condition](https://github.com/openedx/frontend-app-discussions/blob/master/src/discussions/common/AuthorLabel.jsx#L47).

**Comment by awais-ansari** [comment]
@sohailfatima Please rebase the branch. The current branch has some conflicts.

**Comment by mphilbrick211** [comment]
Hi @sohailfatima! I'm just following-up to see if you've had a chance to submit a [CLA form](https://openedx.atlassian.net/wiki/spaces/COMM/pages/941457737/How+to+start+contributing+to+the+Open+edX+code+base)?

File: src/discussions/common/AuthorLabel.test.jsx [hunk 2]
```diff
@@ -90,6 +90,15 @@ describe('Author label', () => {
       },
     );
 
+    it(
+      'it is not clickable when enableInContextSidebar is true',
```
Comment (ayesha-waris) [inline src/discussions/common/AuthorLabel.test.jsx:94]: can be written in previous line

**Review by ayesha-waris — CHANGES_REQUESTED** [review]

**Review by awais-ansari — APPROVED** [review]
Just a best practice as Ayesha mentioned other than LGTM.

**Comment by ayesha-waris** [comment]
fixed in [PR](https://github.com/openedx/frontend-app-discussions/pull/new/Ayesha/INF-1224)

**Comment by openedx-webhooks** [comment]
@sohailfatima Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-app-discussions&entry.867055334=https://github.com/openedx/frontend-app-discussions/pull/653&entry.1484655318=https://github.com/sohailfatima&entry.752974735=2024-01-24+17:47&entry.1917517419=2024-01-31+12:07&entry.2133058324=No) so we can improve your experience in the future.

**Pull request closed without merging**